internal/convert.go

```diff
@@ -168,27 +168,52 @@ func (conv *Conv) SetDataMode() {
 	conv.mode = dataOnly
 }
 
+func dequeue(queue []string) []string {
+	return queue[1:]
+}
+
 // GetDDL Schema returns the Spanner schema that has been constructed so far.
-// Return DDL in alphabetical table order.
+// We sort DDL in alphabetical order, but we maintain the order of interleaved
+// relation by passing tables through a queue
 func (conv *Conv) GetDDL(c ddl.Config) []string {
 	var tables []string
+	var tablesFk []string
 	for t := range conv.SpSchema {
 		tables = append(tables, t)
+		tablesFk = append(tablesFk, t)
 	}
 	sort.Strings(tables)
 	var ddl []string
-	for _, t := range tables {
-		ddl = append(ddl, conv.SpSchema[t].PrintCreateTable(c))
+	printed := make(map[string]bool)
+	for len(tables) > 0 {
+		t := tables[0]
+		if conv.SpSchema[t].InterleaveInto == "" {
```

Comment:
The two blocks of code at lines 191-194 and 197-200 are identical.
Suggest combining them. Maybe something like:

_, found := printed[conv.SpSchema[t].InterleaveInto]
// Print table t if either:
// a) t is not interleaved in another table, or
// b) t is interleaved in another table and that table is already printed.
if conv.SpSchema[t].InterleaveInto == "" || found {
  ... existing code ...
}

Reply:
updated code as mentioned.